regression/cbmc/Pointer_byte_extract10/main.c

```diff
@@ -0,0 +1,45 @@
+void *malloc(__CPROVER_size_t);
+
+typedef union
+{
+  int a;
+  int b;
+} Union;
+
+#ifdef __GNUC__
+typedef struct
+{
+  int Count;
+  Union List[1];
+} __attribute__((packed)) Struct3;
+#else
+typedef struct
+{
+  int Count;
+  Union List[1];
+} Struct3;
+#endif
+
+int main()
+{
+  Struct3 *p = malloc (sizeof (int) + 2 * sizeof(Union));
+  p->Count = 3;
+  int po=0;
+
+  // this should be fine
+  p->List[0].a = 555;
+
+  __CPROVER_assert(p->List[0].b==555, "p->List[0].b==555");
+  __CPROVER_assert(p->List[0].a==555, "p->List[0].a==555");
+
+  // this should be fine
+  p->List[1].b = 999;
```

Comment:
Maybe?  I agree (modulo the change above) that this is in the dynamically allocated region, but I don't know that it is in inside the array and I don't know exact A. which we check and B. the standard would use for determining undefined behaviour.

Reply:
I'm not sure this is covered by the C standard. As far as I recall this is an example from SV-COMP that originated from the Linux kernel. Using a proper flexible array would work, but I'm not sure it would expose the bug that this test was originally mean to expose (which is that we need to look at the actually allocated object instead of the type).

Reply:
Ummm... I'm unsure.  At the risk of making this "what exactly is the semantics of ..." week -- what is it that bounds check is checking?  Is:

```struct s {
  int inputBuffer[LENGTH];
  int outputBuffer[LENGTH];
}

x = inputBuffer[LENGTH];```

An error?